src/Executables/Game/World/Entities/PlayerEntity.cs

```diff
@@ -269,6 +272,26 @@ private void CalculateDefence()
             // todo add defence bonus from quests
         }
 
+        private void CalculateMovement()
+        {
+            MovementSpeed = 150;
+            float modifier = 0;
+            foreach (var slot in Enum.GetValues<EquipmentSlots>())
+            {
+                var item = Inventory.EquipmentWindow.GetItem(slot);
+                if (item == null) continue;
+                var proto = _itemManager.GetItem(item.ItemId);
+                if (proto?.Type != (byte)EItemType.Armor) continue;
+
+                if (proto.Applies[0].Type != (byte) EApplyType.MovSpeed) continue;
```

Comment:
I would do something like this:
```suggestion
                var moveSpeedFromItem = proto.Applies.FirstOrDefault(apply => apply.Type == (byte)EApplyType.MovSpeed);
                if (moveSpeedFromItem is null) continue;
```